File: test10/equi_filter.py
#!/usr/bin/env python3

import sys, re

for line in sys.stdin:
   words = re.split('\s+', line)
   out = []

   for word in words:
      count = {}
      lword = word.lower()

      for letter in lword:
         n = count[letter] = count.get(letter, 0) + 1

      toPrint = 1

      for letter in count.keys():
         if count[letter] != n:
            toPrint = 0

      if toPrint:
         out.append(word)

   print(' '.join(out))

File: test10/equi_filter.pl
#!/usr/bin/perl

foreach $line (<STDIN>) {
   @words = split /\s+/, $line;
   undef @toPrint;
   
   #print join ' ', @words;
   
   foreach $word (@words) {
      $lword = lc $word;
      undef %count;
      
      for $i (0..length($lword) - 1) {
         $char = substr($lword, $i, 1);
         $count{$char}++;
         $n = $count{$char};
      }
      
      #print "word = $lword, n = $n\n";
      #print %count, "\n";
      undef $not_equi;
      
      for $char (keys %count) {
         if ($count{$char} != $n) {
            $not_equi = 1;
         }
      }
      
      if (!$not_equi) {
         push @toPrint, $word;
      }
   }
   
   print join(' ', @toPrint), "\n";
}
